### DISCUSSION TITLE
How can I sync only the content of an online directory on my Onedrive to my local 'sync_dir'

### DISCUSSION THREAD

**Discussion by kdeneve** [abraunegg/onedrive#2655] [Q&A] [answered]
Hello
My sync_dir path on my local system resides at /mnt/Data
I have a directory Data on my onedrive (online)
What I want to achieve is to sync all files and directories that are on the Data folder in ondrive(online) directly under /mnt/Data on my local system
When I do a sync now I have the following: /mnt/Data/Data/with alls subfolder and files from Data
What I want is /mnt/Data/ with all subfolder and files from Data -> on my local system.

I already tried in the sync_list file following includes combinations:
Data
Data/*
Data/asterisk dot asterisk

Can someone help me with this issue please?

Second question:
I am doing all this on a FreeBSD machine because I use a jail on Truenas.
Is there somewere a guide how to use onedrive as a service.
systemctl is not used on FreeBSD ( I am not so familiar with FreeBSD)

Two-way sync on Truenas with onedrive does not seems to be easy, unlike with my QNAP Nas. :-(

Best Regards
Kurt

**Comment by abraunegg** [comment, marked as answer]
@kdeneve 
>When I do a sync now I have the following: /mnt/Data/Data/with alls subfolder and files from Data
>What I want is /mnt/Data/ with all subfolder and files from Data -> on my local system.

How to do this is explained in detail here: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#upload-data-from-the-local-onedrive-folder-to-a-specific-location-on-onedrive . Massage the details as necessary and use `--dry-run` to validate.

> I am doing all this on a FreeBSD machine because I use a jail on Truenas.
> Is there somewere a guide how to use onedrive as a service.
> systemctl is not used on FreeBSD ( I am not so familiar with FreeBSD)

FreeBSD uses the traditional BSD-style init

Please look at https://github.com/abraunegg/onedrive/tree/master/contrib/init.d for an init reference 

You may need to massage this script (it was written for RHEL 6 | CentOS 6 and older non systemd Linux systems) by reading https://docs.freebsd.org/en/articles/rc-scripting/
